compiler/lib/Conversion/LcclToByre/LcclToByre.cpp

```diff
@@ -31,6 +31,27 @@ using namespace llvm;
 
 namespace {
 
+template <typename T>
+ArrayAttr getByreMemoryEffectForLcclOps(T op, PatternRewriter &rewriter) {
```

Comment:
Seems this function is not a specific function only for lccl ops. Could you find or create a general util function?